client/wp-admin-scripts/onboarding-tax-notice/index.js

```diff
@@ -28,30 +27,36 @@ const saveCompleted = () => {
 const showTaxCompletionNotice = () => {
 	const saveButton = document.querySelector( '.woocommerce-save-button' );
 
-	if ( saveButton.classList.contains( 'is-clicked' ) ) {
+	if ( saveButton.classList.contains( 'has-tax' ) ) {
 		return;
 	}
 
-	saveButton.classList.add( 'is-clicked' );
-	saveCompleted().then( () =>
-		dispatch( 'core/notices' ).createSuccessNotice(
-			__( "You've added your first tax rate!", 'woocommerce-admin' ),
-			{
-				id: 'WOOCOMMERCE_ONBOARDING_TAX_NOTICE',
-				actions: [
-					{
-						url: getAdminLink( 'admin.php?page=wc-admin' ),
-						label: __( 'Continue setup.', 'woocommerce-admin' ),
-					},
-				],
-			}
-		)
-	);
+	saveCompleted().then( () => {
+		// Check if a row was added successfully after WooCommerce removes invalid rows.
+		if ( ! document.querySelector( '.tips' ) ) {
+			return;
+		}
+		if ( saveButton.disabled ) {
```

Comment:
Thanks for removing the `setTimeout`, and replacing it with this.
My apologies if I was kinda vague with the `saveButton.disabled` suggestion. I meant to update the `saveCompleted` method with using the disabled state like this:
```
const saveCompleted = ( saveButton ) => {
	if ( saveButton && ! saveButton.disabled ) {
		const promise = new Promise( ( resolve ) => {
			window.requestAnimationFrame( resolve );
		} );
		return promise.then( () => saveCompleted( saveButton ) );
	}

	return Promise.resolve( true );
};
```
You did have to pass the saveButton in on line 34 -> `saveCompleted(saveButton).then(...`
Currently it does this check `document.querySelector( '.blockUI.blockOverlay' )` which I don't think is relevant anymore.

Reply:
Thank you for the feedback and sorry that I completely miss read it! The changes have been pushed.